tools/sync_test_server/ros-testing-server.js

```diff
@@ -40,50 +41,65 @@ function startRealmObjectServer(done) {
-                    { env: env});
-            // local config:
-            syncServerChildProcess.stdout.on('data', (data) => {
-                if (logFindingCounter != 0 && /client: Closing Realm file: .*__auth.realm/.test(data)) {
-                    if (logFindingCounter == 1) {
-                        done()
+    stopRealmObjectServer(function(err) {
+        if(err) {
+          return;
+        }
+        temp.mkdir('ros', function(err, path) {
+            if (!err) {
+                var oldCwd = process.cwd();
+                process.chdir(path);
+                syncServerDir = path;
+                winston.info("Starting sync server in ", path);
+                var env = Object.create( process.env );
+                winston.info(env.NODE_ENV);
+                env.NODE_ENV = 'development';
+                syncServerChildProcess = spawn('realm-object-server',
+                        ['--root', path,
+                        '--configuration', '/configuration.yml'],
+                        { env: env});
+                // local config:
+                syncServerChildProcess.stdout.on('data', (data) => {
+                    if (logFindingCounter != 0 && /client: Closing Realm file: .*__auth.realm/.test(data)) {
+                        if (logFindingCounter == 1) {
+                            done()
+                        }
+                        logFindingCounter--
                     }
-                    logFindingCounter--
-                }
-                winston.info(`stdout: ${data}`);
-            });
+                    winston.info(`stdout: ${data}`);
+                });
 
-            syncServerChildProcess.stderr.on('data', (data) => {
-                winston.info(`stderr: ${data}`);
-            });
+                syncServerChildProcess.stderr.on('data', (data) => {
+                    winston.info(`stderr: ${data}`);
+                });
 
-            syncServerChildProcess.on('close', (code) => {
-                winston.info(`child process exited with code ${code}`);
-            });
-        }
+                syncServerChildProcess.on('close', (code) => {
+                    winston.info(`child process exited with code ${code}`);
+                });
+                process.chdir(oldCwd);
+            }
+        });
     });
 }
 
-function stopRealmObjectServer() {
+function stopRealmObjectServer(callback) {
     if (syncServerChildProcess) {
-        syncServerChildProcess.kill();
-        syncServerChildProcess = null;
-        exec('rm -r ' + 'realm-object-server', function (err, stdout, stderr) {
-            if (err) {
-                winston.error(err);
-            } else {
-                winston.info("realm-object-server directory deleted");
-            }
+        syncServerChildProcess.on('exit', function() {
+            syncServerChildProcess = null;
+            exec('rm -r ' + syncServerDir, function (err, stdout, stderr) {
```

Comment:
Since those dirs would never be deleted, I would rather avoid this for now. Mostly for selfish reasons, since I'm battling disk space daily, but also because I'm not sure we generally need that level of debugging?